packages/application-components/src/components/internals/page-top-bar.tsx

```diff
@@ -0,0 +1,63 @@
+import { css } from '@emotion/react';
+import { useIntl } from 'react-intl';
+import { ListIcon } from '@commercetools-uikit/icons';
+import FlatButton from '@commercetools-uikit/flat-button';
+import { customProperties } from '@commercetools-uikit/design-system';
+import type { MouseEvent, KeyboardEvent } from 'react';
+import messages from './messages';
+
+// NOTE: the `MessageDescriptor` type is exposed by `react-intl`.
+// However, we need to explicitly define this otherwise the prop-types babel plugin
+// does not recognize the object shape.
+type MessageDescriptor = {
+  id: string;
+  description?: string | object;
+  defaultMessage?: string;
+};
```

Comment:
I think we should also add the `values` property to allow for messages that contain parameters as we had to do in ui-kit.

```suggestion
type MessageDescriptor = {
  id: string;
  description?: string | object;
  defaultMessage?: string;
  values?; Record<string, React.ReactNode>
};
```